### PR TITLE
Upgrade substrate dependency to match 0.9.24

### PR DESCRIPTION
**Pull request by sudipghimire533**
All substrate-related dependency have been updated to v0.9.24. 

Remaining changes:
In `pallet_bounties` there are new types in `Config` trait namely: `CuratorDepositMin` & `CuratorDepositMax`. Proper true value is to be assigned here

### CODE REVIEW

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -501,6 +503,8 @@ parameter_types! {
     pub const MaximumReasonLength: u32 = 300;
     pub const BountyValueMinimum: Balance = 5 * DOLLARS;
     pub const BountyCuratorDeposit: Permill = Permill::from_percent(50);
+    pub const BountyCuratorDepositMin: Balance = 10 * DOLLARS;
+    pub const BountyCuratorDepositMax: Balance = 100 * DOLLARS;
```
Comment (sudipghimire533) [inline runtime/src/lib.rs]: Put true value here

**Review by ayushmishra2005 — COMMENTED** [review]
@sudipghimire533 Please address below comments:

1) Replace https://github.com/DataHighway-DHX/DataHighway-Parachain/blob/sg/upgrade-polkadot-0.9.24/res/kusama/kusama.json  with latest Kusama 0.9.24 spec. You can find here https://github.com/paritytech/polkadot/blob/v0.9.24/node/service/res/kusama.json

2) Same for rococo

3) Please test it on local by connecting with local relaychain

4) Try to connect with live Kusama and Rococo

**Comment by sudipghimire533** [comment]
How file was updated in `res/` folder:
- **kusama/kusama.json** & **rococo/rococo.json** from [here](https://github.com/paritytech/polkadot/tree/v0.9.24/node/service/res)
- **kusama/datahighway-collator.gz** & **rococo/datahighway-collator.gz**: compressed binary. from `target/release/dataighway-collator` in workspace
- **rococo/rococo-parachain-raw.json**: `datahighway-collator --chain rococo --raw --disable-default-bootnode`
- **kusama/kusama-parachain-raw.json**: `datahighway-collator --chain kusama --raw --disable-default-bootnode`
- **kusama/\*.wasm** & **rococo/\*.wasm**: copied from `target/release/wbuild/`
- **rococo/rococo-parachain-state**: `datahighway-collator export-genesis-state --chain rococo-parachain-raw.json`
- **kusama/kusama-parachain-state**: `datagighway-collator export-genesis-state --chain kusama-parachain-raw.json`
- **rococo/rococo-parachain-wasm**: `datahighway-collator export-genesis-wasm --chain rococo-parachain-raw.json`
- **rococo/kusama-parachain-wasm**: `datahighway-collator export-genesis-wasm --chain kusama-parachain-raw.json`

File: runtime/src/lib.rs [hunk 2]
```diff
@@ -384,7 +385,8 @@ parameter_types! {
 
 impl pallet_transaction_payment::Config for Runtime {
     type OnChargeTransaction = pallet_transaction_payment::CurrencyAdapter<Balances, DealWithFees>;
-    type TransactionByteFee = TransactionByteFee;
+    // type TransactionByteFee = TransactionByteFee;
```
Comment (ayushmishra2005) [inline runtime/src/lib.rs]: Please remove this

**Review by ayushmishra2005 — CHANGES_REQUESTED** [review]
Let's not modify any file under res/kusama and res/rococo except `kusama.json` and `rococo.json`

File: node/Cargo.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = 'datahighway-collator'
-version = '3.6.0'
+version = '3.6.1'
```
Comment (ayushmishra2005) [inline node/Cargo.toml:3]: You may need to do same in runtime.

**Review by ayushmishra2005 — CHANGES_REQUESTED** [review]

File: runtime/src/lib.rs (hunk 1 above)
Comment (ayushmishra2005) [reply runtime/src/lib.rs]: What does this mean?

File: runtime/src/lib.rs [hunk 4]
```diff
@@ -541,9 +544,11 @@ impl pallet_bounties::Config for Runtime {
     type BountyDepositBase = BountyDepositBase;
     type BountyDepositPayoutDelay = BountyDepositPayoutDelay;
     type BountyUpdatePeriod = BountyUpdatePeriod;
-    type BountyCuratorDeposit = BountyCuratorDeposit;
     type BountyValueMinimum = BountyValueMinimum;
     type DataDepositPerByte = DataDepositPerByte;
+    type CuratorDepositMin = BountyCuratorDepositMin; // TODO: put correct value here
+    type CuratorDepositMax = BountyCuratorDepositMax; // TODO: put correct value here
+    type CuratorDepositMultiplier = BountyCuratorDeposit; // TODO: put correct value here
```
Comment (ayushmishra2005) [inline runtime/src/lib.rs]: Please confirm these values from team and remove TODO comments after confirmation.

File: runtime/src/lib.rs [hunk 5]
```diff
@@ -689,6 +693,7 @@ impl pallet_recovery::Config for Runtime {
     type FriendDepositFactor = FriendDepositFactor;
     type MaxFriends = MaxFriends;
     type RecoveryDeposit = RecoveryDeposit;
+    type WeightInfo = ();
```
Comment (ayushmishra2005) [inline runtime/src/lib.rs]: Can't we use something like this `pallet_recovery::weights::SubstrateWeight<Runtime>;` instead of `()`

**Review by ayushmishra2005 — CHANGES_REQUESTED** [review]

File: runtime/src/lib.rs (hunk 4 above)
Comment (sudipghimire533) [reply runtime/src/lib.rs]: Researching...

File: runtime/src/lib.rs (hunk 1 above)
Comment (sudipghimire533) [reply runtime/src/lib.rs]: I am Researching....

File: runtime/src/lib.rs (hunk 5 above)
Comment (sudipghimire533) [reply runtime/src/lib.rs]: We can indeed. Done

**Review by ayushmishra2005 — APPROVED** [review]

**Pull request merged**